diff --git a/sshfs.c b/sshfs.c
index e253291..ae4955a 100644
--- a/sshfs.c
+++ b/sshfs.c
@@ -3710,6 +3710,15 @@ static int sshfs_opt_proc(void *data, const char *arg, int key,
 				&& ':' == arg[1] && '\0' == arg[2]) {
 				/* drive: make a copy */
 				sshfs.mountpoint = strdup(arg);
+			} else
+			if ('\\' == arg[0] && '\\' == arg[1] &&
+			    ('.' == arg[2] || '?' == arg[2]) &&
+			    '\\' == arg[3] && (
+					('A' <= arg[4] && arg[4] <= 'Z') ||
+					('a' <= arg[4] && arg[4] <= 'z')
+				) &&
+				':' == arg[5] && '\0' == arg[6]) {
+				sshfs.mountpoint = strdup(arg);
 			} else {
 				/* path: split into dirname, basename and check dirname */
 				char *dir;
